integration/otlp_ingestion_test.go

```diff
@@ -71,9 +79,34 @@ func TestOTLPIngestion(t *testing.T) {
 	require.Equal(t, model.ValMatrix, rangeResult.Type())
 	require.Equal(t, expectedMatrix, rangeResult.(model.Matrix))
 
+	// Query the metadata
+	metadataResult, err := c.GetPrometheusMetadata("", 0)
+	require.NoError(t, err)
+	require.Equal(t, 200, metadataResult.StatusCode)
+
+	metadataResponseBody, err := io.ReadAll(metadataResult.Body)
+	require.NoError(t, err)
+
+	expectedJSON := `
+	{
+	   "status":"success",
+	   "data":{
+		  "series_1":[
+			 {
+				"type":"counter",
+				"help":"foo",
+				"unit":"foo"
+			 }
+		  ]
+	   }
+	}
+	`
+
+	require.JSONEq(t, expectedJSON, string(metadataResponseBody))
+
 	// Push series with histograms to Mimir
 	series, expectedVector, _ = generateHistogramSeries("series", now)
-	res, err = c.PushOTLP(series)
+	res, err = c.PushOTLP(series, metadata)
```

Comment:
Might be worth adding a check that the metadata is stored correctly for this histogram series too.

Reply:
Added.